### PR TITLE
Implement jacobian descent for pre-training

### PR DESCRIPTION
**Pull request by Reid-Taylor**
- Implemented Jacobian Descent via UPGrad with TorchJD as a drop-in replacement for the previously implemented sum of gradients. 
- Configured weighting still applies to the initial gradients. 
- Rewrote one test to spin up an optimizer engine stub to enable parity with previously tested training step output.
- Added relevant pytests to root to assert that the JD engine is rebuilt after modifying schema and loading model from checkpoint

### CODE REVIEW

**Comment by Reid-Taylor** [comment]
Closing; will refactor some pieces and fix an emergent error with DDP prior to the next PR for this feature enhancement
